lib/Sema/CSDiag.cpp

```diff
@@ -1816,7 +1840,11 @@ enum TCCFlags {
   /// Re-type-check the given subexpression even if the expression has already
   /// been checked already.  The client is asserting that infinite recursion is
   /// not possible because it has relaxed a constraint on the system.
-  TCC_ForceRecheck = 0x02
+  TCC_ForceRecheck = 0x02,
+    
+  /// tell typeCheckExpression that it is ok to produce an ambiguous result,
+  /// it can just fill in holes with UnresolvedType and we'll deal with it.
+  TCC_AllowUnresolvedTypeVariables = 0x04
```

Comment:
This is just plumbing to pass down through a couple layers of calls to set the pre-existing TypeCheckExprFlags::AllowUnresolvedTypeVariables.